ambry-store/src/test/java/com.github.ambry.store/CuratedLogIndexState.java

```diff
@@ -393,6 +474,119 @@ void verifyEntriesForHardDeletes(Set<MockId> deletedIds) throws IOException {
   }
 
   /**
+   * Returns {@code true} if the {@code id} is considered deleted at {@code referenceTimeMs}.
+   * @param id the {@link MockId} to check.
+   * @param referenceTimeMs the reference time in ms at which the check needs to be performed.
+   * @return {@code true} if the {@code id} is considered deleted at {@code referenceTimeMs}. {@code false} otherwise.
+   */
+  boolean isDeletedAt(MockId id, long referenceTimeMs) {
+    Offset deleteIndexSegmentStartOffset = indexSegmentStartOffsets.get(id).getSecond();
+    return deleteIndexSegmentStartOffset != null
```

Comment:
The id passed may not be deleted. It is fair to call `isDeletedAt()` on keys with no delete records.